state/pruner.go

```diff
@@ -205,30 +207,42 @@ func (p *Pruner) GetABCIResRetainHeight() (int64, error) {
 	return p.stateStore.GetABCIResRetainHeight()
 }
 
-func (p *Pruner) pruningRoutine() {
+func (p *Pruner) pruneABCIResRoutine() {
+	p.logger.Info("Started pruning ABCI results", "interval", p.interval.String())
+	lastABCIResRetainHeight := int64(0)
+	for {
+		select {
+		case <-p.Quit():
+			return
+		default:
+			newABCIResRetainHeight := p.pruneABCIResToRetainHeight(lastABCIResRetainHeight)
+			p.observer.PrunerPruned(&PrunedInfo{
```

Comment:
The observer's `PrunerPruned` method would need to be split into two methods:
- `PrunerPrunedBlocks(*BlocksPrunedInfo)`
- `PrunerPrunedABCIResults(*ABCIResponsesPrunedInfo)`

If we split the pruning routines, it no longer makes sense to have a single event represent both.

Reply:
Done!